src/swap/components/SwapAmountInput.tsx

```diff
@@ -64,6 +63,14 @@ export function SwapAmountInput({
   const hasInsufficientBalance =
     type === 'from' && Number(source.balance) < Number(source.amount);
 
+  const formatUSD = (amount: string) => {
+    if (!amount) {
+      return '~$0.00';
+    }
+    const roundedAmount = Number(getRoundedAmount(amount, 2) || 0);
```

Comment:
shouldn't getRoundedAmount always return a value?   when would we hit the 0 fallback?

Reply:
Good callout, updated!